[build] MacOS incremental builds broken

## Environment

* SuperCollider version: current dev/3.10
* Operating system: macOS 10.14.6
* Other details (Qt version, audio driver, etc.): Xcode 11.0, CMake 3.15.3

## Steps to reproduce

1. build the SuperCollider target once
2. modify a file in the SC IDE target, forcing it to rebuild
3. rebuild the SuperCollider target

## Expected vs. actual behavior

instead of the build completing normally, it appears that the sclang target already existing in the app package causes the (AFAICT, unavoidable) codesigning after building the package to fail. here's the full output:

```
=== BUILD TARGET SuperCollider OF PROJECT SuperCollider WITH CONFIGURATION Debug ===

Check dependencies

Ld build/editors/sc-ide/Debug/SuperCollider.app/Contents/MacOS/SuperCollider normal x86_64
    cd /Users/brianheim/git/sc2
    /Users/brianheim/git/sc2/build/launch-cxx -target x86_64-apple-macos10.14 -isysroot /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.15.sdk -L/Users/brianheim/git/sc2/build/editors/sc-ide/Debug -F/Users/brianheim/git/sc2/build/editors/sc-ide/Debug -F/usr/local/opt/qt/lib -filelist /Users/brianheim/git/sc2/build/editors/sc-ide/SuperCollider.build/Debug/SuperCollider.build/Objects-normal/x86_64/SuperCollider.LinkFileList -Xlinker -object_path_lto -Xlinker /Users/brianheim/git/sc2/build/editors/sc-ide/SuperCollider.build/Debug/SuperCollider.build/Objects-normal/x86_64/SuperCollider_lto.o -Xlinker -no_deduplicate -stdlib=libc++ -Wl,-search_paths_first -Wl,-headerpad_max_install_names -stdlib=libc++ /Users/brianheim/git/sc2/build/editors/sc-ide/Debug/libscide.a /usr/local/opt/qt/lib/QtConcurrent.framework/QtConcurrent /usr/local/opt/qt/lib/QtSql.framework/QtSql /usr/local/opt/qt/lib/QtWebEngineWidgets.framework/QtWebEngineWidgets /usr/local/opt/qt/lib/QtPrintSupport.framework/QtPrintSupport /usr/local/opt/qt/lib/QtWidgets.framework/QtWidgets /usr/local/opt/qt/lib/QtWebEngineCore.framework/QtWebEngineCore /usr/local/opt/qt/lib/QtQuick.framework/QtQuick /usr/local/opt/qt/lib/QtWebChannel.framework/QtWebChannel /usr/local/opt/qt/lib/QtQml.framework/QtQml /usr/local/opt/qt/lib/QtGui.framework/QtGui /usr/local/opt/qt/lib/QtPositioning.framework/QtPositioning /usr/local/opt/qt/lib/QtWebSockets.framework/QtWebSockets /usr/local/opt/qt/lib/QtNetwork.framework/QtNetwork /usr/local/opt/qt/lib/QtCore.framework/QtCore /Users/brianheim/git/sc2/build/external_libraries/Debug/libyaml.a /Users/brianheim/git/sc2/build/external_libraries/Debug/liboscpack.a /Users/brianheim/git/sc2/build/external_libraries/Debug/libboost_system_lib.a /Users/brianheim/git/sc2/build/external_libraries/Debug/libboost_filesystem_lib.a -framework CoreServices -framework Foundation -framework Cocoa /usr/lib/libpthread.dylib -Xlinker -dependency_info -Xlinker /Users/brianheim/git/sc2/build/editors/sc-ide/SuperCollider.build/Debug/SuperCollider.build/Objects-normal/x86_64/SuperCollider_dependency_info.dat -o /Users/brianheim/git/sc2/build/editors/sc-ide/Debug/SuperCollider.app/Contents/MacOS/SuperCollider

Touch build/editors/sc-ide/Debug/SuperCollider.app
    cd /Users/brianheim/git/sc2
    /usr/bin/touch -c /Users/brianheim/git/sc2/build/editors/sc-ide/Debug/SuperCollider.app

CodeSign build/editors/sc-ide/Debug/SuperCollider.app
    cd /Users/brianheim/git/sc2

Signing Identity:     "-"

    /usr/bin/codesign --force --sign - --entitlements /Users/brianheim/git/sc2/build/editors/sc-ide/SuperCollider.build/Debug/SuperCollider.build/SuperCollider.app.xcent --timestamp=none /Users/brianheim/git/sc2/build/editors/sc-ide/Debug/SuperCollider.app
/Users/brianheim/git/sc2/build/editors/sc-ide/Debug/SuperCollider.app: code object is not signed at all
In subcomponent: /Users/brianheim/git/sc2/build/editors/sc-ide/Debug/SuperCollider.app/Contents/MacOS/sclang
Command /usr/bin/codesign failed with exit code 1

** BUILD FAILED **


The following build commands failed:
	CodeSign build/editors/sc-ide/Debug/SuperCollider.app
(1 failure)
```

i think (hunch) the problem is that our cmake scripts don't use the right pattern for indicating dependencies among components. sclang depends on SuperCollider (this is backwards) and is copied into the bundle as a post build step. i think there may be a way to indicate that an executable is a resource in an app bundle.

i'm going to debug this by creating a small toy project similar to ours in structure that reproduces the problem. (trying to debug our project directly is probably going to give me a headache, and there's too much noise)

are other people experiencing this? is it just my version of CMake? i can't imagine older versions handled code signing this way.

--- a/.github/workflows/actions.yml
+++ b/.github/workflows/actions.yml
@@ -299,7 +299,7 @@ jobs:
 
           echo "EXTRA_CMAKE_FLAGS:" $EXTRA_CMAKE_FLAGS
 
-          cmake -G"Xcode" -DRULE_LAUNCH_COMPILE=ccache -DCMAKE_OSX_DEPLOYMENT_TARGET=10.10 -DSUPERNOVA=ON $EXTRA_CMAKE_FLAGS -DCMAKE_XCODE_ATTRIBUTE_CODE_SIGNING_REQUIRED="NO" -DCMAKE_XCODE_ATTRIBUTE_CODE_SIGN_IDENTITY="" .. --debug-output
+          cmake -G"Xcode" -DRULE_LAUNCH_COMPILE=ccache -DCMAKE_OSX_DEPLOYMENT_TARGET=10.10 -DSUPERNOVA=ON $EXTRA_CMAKE_FLAGS .. --debug-output
       - name: build
         run: cmake --build $BUILD_PATH --config Release --target install
       - name: create archive

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -243,6 +243,11 @@ endif()
 
 option(SC_ABLETON_LINK "Build with Ableton Link support" ON)
 
+if(APPLE)
+    # codesigning breaks increamental builds in Xcode 11+
+    option(SC_DISABLE_XCODE_CODESIGNING "Disable codesigning in XCode" ON)
+endif()
+
 #############################################
 # some default libraries
 
@@ -390,6 +395,11 @@ endif()
 
 if (NO_GPL3)
 	add_definitions(-DNO_GPL3)
+endif()
+
+if(SC_DISABLE_XCODE_CODESIGNING)
+    set(CMAKE_XCODE_ATTRIBUTE_CODE_SIGNING_REQUIRED "NO")
+    set(CMAKE_XCODE_ATTRIBUTE_CODE_SIGN_IDENTITY "")
 endif()
 
 #############################################

